src/libstore/include/nix/store/uds-remote-store.hh

```diff
@@ -7,24 +7,29 @@
 
 namespace nix {
 
-struct UDSRemoteStoreConfig : virtual LocalFSStoreConfig, virtual RemoteStoreConfig
+struct UDSRemoteStoreConfig :
+    std::enable_shared_from_this<UDSRemoteStoreConfig>,
+    Store::Config,
+    LocalFSStore::Config,
+    RemoteStore::Config
 {
-    // TODO(fzakaria): Delete this constructor once moved over to the factory pattern
-    // outlined in https://github.com/NixOS/nix/issues/10766
-    using LocalFSStoreConfig::LocalFSStoreConfig;
-    using RemoteStoreConfig::RemoteStoreConfig;
+    static config::SettingDescriptionMap descriptions();
+
+    UDSRemoteStoreConfig(const StoreReference::Params & params)
+        : UDSRemoteStoreConfig{"unix", "", params}
+    {}
 
     /**
      * @param authority is the socket path.
      */
     UDSRemoteStoreConfig(
         std::string_view scheme,
         std::string_view authority,
-        const Params & params);
+        const StoreReference::Params & params);
 
-    const std::string name() override { return "Local Daemon Store"; }
+    static const std::string name() { return "Local Daemon Store"; }
 
-    std::string doc() override;
+    static std::string doc();
```

Comment:
Downgrade to constants?

Reply:
We I think we could, but I am tempted to do in a separate commit since this one is already so big. How does that sound?